heft/heft-webpack-basic-tutorial/webpack.config.js

```diff
@@ -12,45 +15,171 @@ function createWebpackConfig({ production }) {
+                  // - a lambda disables CSS modules only if the lambda returns false; the function parameter is
+                  //   the resource path
+                  // - a RegExp disables CSS modules only if the resource path does not match the RegExp
+                  //
+                  // NOTE: Counterintuitively, if you instead set "modules=true" then CSS modules are enabled
+                  //       without magic, equivalent to "auto: () => true" instead of "auto: true"
+                  //
+                  // DEFAULT: "true" (i.e. path based magic)
+                  auto: (resourcePath) => {
+                    // Enable CSS modules unless the filename opts out using a file extension like "filename.global.scss"
+                    return !/\.global\.\w+$/i.test(resourcePath);
+                  },
+
+                  // This setting has no effect unless CSS modules is enabled. Possible values:
+                  // - "local": global CSS by default, overridable using the ":local" selector
+                  // - "global": local CSS by default, overridable using the ":global" selector
+                  // - "pure": requires selectors to contain at least one local class or id
+                  // - a lambda that returns the mode string; the function parameter is the resource path
+                  //
+                  // DEFAULT: "local"
+                  mode: 'local',
+
+                  // Set this to true if you want to be able to reference the global declarations using import statements
+                  // similar to local CSS modules
+                  //
+                  // DEFAULT: false
+                  // exportGlobals: true,
+
+                  // Provide a recognizable class/module names for developers
+                  //
+                  // DEFAULT: "[hash:base64]"
+                  localIdentName: production ? '[hash:base64]' : '[local]__[hash:base64:5]'
+                },
+
+                sourceMap: !production
+              }
+            },
+
+            {
+              // PostCSS is a general-purpose CSS transformer; however, we prefer to avoid custom CSS syntaxes
+              // and only use the standard SASS syntax.  Thus postcss-loader is used here only to apply the popular
+              // "autoprefixer" plugin improves browser compatibility by generating vendor prefixes.
+              // https://www.npmjs.com/package/postcss-loader
+              loader: 'postcss-loader',
+              options: {
+                postcssOptions: {
+                  plugins: [
+                    // https://www.npmjs.com/package/autoprefixer
+                    autoprefixer
+                  ]
+                },
+
+                sourceMap: !production
+              }
+            },
+
+            {
+              // Compiles SASS syntax into CSS
+              // https://www.npmjs.com/package/sass-loader
+              loader: 'sass-loader',
```

Comment:
@bartvandenende-wm pointed out that this recipe will apply SASS transformations to .css files, which we don't want

Reply:
We're thinking to change the rig to work like this:

- `.css` : NO sass, NO autoprefixer, NO modules
- `.scss` : sass, autoprefixer, modules
- `.global.scss` : sass, autoprefixer, NO modules

...and if someone wants CSS with modules, they can simply use the `.sass` file extension, since SASS is basically a superset of CSS.

There's a small performance penalty for applying SASS to a plain CSS file, but @halfnibble pointed out that the extra syntax validation probably justifies that cost.

Reply:
This was implemented in ddf34ea4f3a1b6f75bab74a42bf47896bc807e85